In [3]:
pip install mglearn

     |████████████████████████████████| 540 kB 5.5 MB/s 
  Created wheel for mglearn: filename=mglearn-0.1.9-py2.py3-none-any.whl size=582637 sha256=eb7675483b0573e8df5fae7a41f4c37185109c7768c3897e9d948231b9fe9314
  Stored in directory: /root/.cache/pip/wheels/f1/17/e1/1720d6dcd70187b6b6c3750cb3508798f2b1d57c9d3214b08b
Successfully built mglearn


In [4]:
from IPython.display import display
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd
import mglearn

In [7]:
from google.colab import files
uploaded = files.upload()

Saving titanic.csv to titanic.csv


In [8]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [9]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [10]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


In [12]:
import io
titanic = pd.read_csv(io.BytesIO(uploaded['titanic.csv']))

In [13]:
titanic

PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]

In [20]:
data = pd.read_csv(io.BytesIO(uploaded['titanic.csv']))

In [19]:
print(titanic.Survived.value_counts())

0    549
1    342
Name: Survived, dtype: int64


# **1. 타이타닉호에 승선했던 승객은 누구인가?**

In [21]:
titanic.info

<bound method DataFrame.info of      PassengerId  Survived  Pclass  ...     Fare Cabin  Embarked
0              1         0       3  ...   7.2500   NaN         S
1              2         1       1  ...  71.2833   C85         C
2              3         1       3  ...   7.9250   NaN         S
3              4         1       1  ...  53.1000  C123         S
4              5         0       3  ...   8.0500   NaN         S
..           ...       ...     ...  ...      ...   ...       ...
886          887         0       2  ...  13.0000   NaN         S
887          888         1       1  ...  30.0000   B42         S
888          889         0       3  ...  23.4500   NaN         S
889          890         1       1  ...  30.0000  C148         C
890          891         0       3  ...   7.7500   NaN         Q

[891 rows x 12 columns]>

--> 총 891명의 승객이 승선. 승객의 클래스(Pclass)는 1~3등급으로 나뉘어짐. 부모/형제와 같이 승선(Sibsp, Parch) 한 경우도 있으며, 총 3군데(S,C,Q)에서 승선(Embarked)한 것으로 추정

# **2. 승객은 어디서 묵었으며(Cabin), 그들의 class와 어떤 관계가 있는가?**

In [24]:
#data.columns

In [36]:
df = pd.read_csv("titanic.csv", index_col=0)
df.head(10)

Survived  Pclass  ... Cabin Embarked
PassengerId                    ...               
1                   0       3  ...   NaN        S
2                   1       1  ...   C85        C
3                   1       3  ...   NaN        S
4                   1       1  ...  C123        S
5                   0       3  ...   NaN        S
6                   0       3  ...   NaN        Q
7                   0       1  ...   E46        S
8                   0       3  ...   NaN        S
9                   1       3  ...   NaN        S
10                  1       2  ...   NaN        C

[10 rows x 11 columns]

In [31]:
df.dropna()

Survived  Pclass  ...        Cabin Embarked
PassengerId                    ...                      
2                   1       1  ...          C85        C
4                   1       1  ...         C123        S
7                   0       1  ...          E46        S
11                  1       3  ...           G6        S
12                  1       1  ...         C103        S
...               ...     ...  ...          ...      ...
872                 1       1  ...          D35        S
873                 0       1  ...  B51 B53 B55        S
880                 1       1  ...          C50        C
888                 1       1  ...          B42        S
890                 1       1  ...         C148        C

[183 rows x 11 columns]

# **3. 어떤 요소가 승객들을 생존하게 만들었는가?**
  - 침몰한 타이타닉호에서 어떤 요소가 생존 확률을 높였는지 분석.
  - 생존한 승객의 정보는 Survived에 0과 1로 표기

In [38]:
#fig, axes = plt.subplots(4,3, figsize=(10,20))
#survived = titanic.data[titanic.survived == 1]
#died = titanic.data[titanic.survived == 0]

ax = axes.ravel()

for i in range(12):
    _, bins = np.histogram(titanic.data[:,i], bins=50)
    ax[i].his(survived[:,i], bins=bins, color=survived.cm3(0), alpha=.5)
    ax[i].his(died[:,i], bins=bins, color=survived.cm3(2), alpha=.5)
    ax[i].set_title(titanic.feature_names[i])
    ax[i].set_yticks(())
ax[0].set_xlabel("Size of Features")    
ax[0].set_ylabel("Numbers")
ax[0].legend(["Survived", "died"], loc="best")
fig.tight_layout()


AttributeError: ignored

# **4. 생존자는 몇 명인지 그래프로 확인.**

In [43]:
sv=titanic.groupby('Sex')['Survived'].sum()
sv

Sex
female    233
male      109
Name: Survived, dtype: int64

In [ ]:
print(titanic.Survived.value_counts())

0    549
1    342
Name: Survived, dtype: int64


--> 남자 109명, 여자 233명 생존, 총 342명 생존

# **5. 생존 확률에 어떤 요소가 작용했는지 알아보기 위해 Pclass Feature에 대해 다른 속성들과의 상관관계가 있는지 2개 이상 속성으로 상관관계를 시각화를 통해 도출.**